webui/src/app/lib/actions.ts

```diff
@@ -1,37 +1,66 @@
-/* 
+/*
  * Licensed to the Apache Software Foundation (ASF) under one
  * or more contributor license agreements.  See the NOTICE file
  * distributed with this work for additional information
  * regarding copyright ownership.  The ASF licenses this file
  * to you under the Apache License, Version 2.0 (the
  * "License"); you may not use this file except in compliance
  * with the License.  You may obtain a copy of the License at
- * 
+ *
  *   http://www.apache.org/licenses/LICENSE-2.0
- * 
+ *
  * Unless required by applicable law or agreed to in writing,
  * software distributed under the License is distributed on an
  * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
  * KIND, either express or implied.  See the License for the
  * specific language governing permissions and limitations
- * under the License. 
+ * under the License.
  */
-'use server';
+"use server";
```

Comment:
Can you move the server part to the client? We need to compile it to a static frontend file.
In our scenario, Client-side rendering is OK; we don't need the runtime features of Next.js.

Reply:
should I move api.ts (and some other server components) logic also to client side as it depends on config.yaml?

Reply:
Yes, We should not depend on backend config. If necessary, we can add frontend config.

We can use relative paths to access the backend. This way, we won't need to depend on `config.yml`. 
During development, we can use Next.js to proxy our requests to the local backend at `127.0.0.1:9379`.
https://dev.to/george3447/setting-up-proxy-in-nextjs-for-dev-enviornment-1nh3

Reply:
@jihuayu I updated icons to use fontawesome packages, added a proxy for dev environment and changed server components to client side to better suit our usecase.